src/main/java/com/xebialabs/overthere/cifs/winrm/WinRmConnection.java

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright (c) 2008-2016, XebiaLabs B.V., All rights reserved.
+ *
+ *
+ * Overthere is licensed under the terms of the GPLv2
+ * <http://www.gnu.org/licenses/old-licenses/gpl-2.0.html>, like most XebiaLabs Libraries.
+ * There are special exceptions to the terms and conditions of the GPLv2 as it is applied to
+ * this software, see the FLOSS License Exception
+ * <http://github.com/xebialabs/overthere/blob/master/LICENSE>.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms
+ * of the GNU General Public License as published by the Free Software Foundation; version 2
+ * of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin St, Fifth
+ * Floor, Boston, MA 02110-1301  USA
+ */
+package com.xebialabs.overthere.cifs.winrm;
+
+import com.xebialabs.overthere.*;
+import com.xebialabs.overthere.cifs.CifsConnectionType;
+import com.xebialabs.overthere.cifs.ProcessConnection;
+import com.xebialabs.overthere.cifs.WinrmHttpsCertificateTrustStrategy;
+import com.xebialabs.overthere.cifs.WinrmHttpsHostnameVerificationStrategy;
+import com.xebialabs.overthere.spi.AddressPortMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static com.xebialabs.overthere.ConnectionOptions.*;
+import static com.xebialabs.overthere.cifs.CifsConnectionBuilder.*;
+import static com.xebialabs.overthere.cifs.ConnectionValidator.checkIsWindowsHost;
+import static com.xebialabs.overthere.cifs.ConnectionValidator.checkNotOldStyleWindowsDomain;
+import static com.xebialabs.overthere.smb.SmbConnectionBuilder.SMB_PROTOCOL;
+import static com.xebialabs.overthere.util.OverthereUtils.checkArgument;
+import static com.xebialabs.overthere.util.OverthereUtils.checkNotNull;
+import static com.xebialabs.overthere.util.OverthereUtils.closeQuietly;
+import static java.lang.String.format;
+import static java.net.InetSocketAddress.createUnresolved;
+
+/**
+ * A connection to a Windows host using a Java implementation of WinRM.
+ */
+public class WinRmConnection implements ProcessConnection {
```

Comment:
Move each of the ProcessConnection to a `com.xebialabs.overthere.<connection>` package, so not a subpackage of the `cifs` package. This makes the dependency graph nicer, esp. with SMB added. Also move the relevant 'internal' classes with it, and ensure that they're as tight as possible.